### PR TITLE
Pearl Cheat sheet

### PR DESCRIPTION
**Pull request by salmanfarisvp**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`New Pearl Cheat Sheet`**

This is a handy reference, meant for beginning Perl programmers. Not everything is mentioned **
https://duck.co/ia/view/pearl

###### Description of changes

What changes does this PR introduce? 

###### Related Issues or Discussions 

What issues or discussions (if any) does this PR address? Please provide the issue # or link.

###### People to notify

@mention any relevant people/organizations

---

Instant Answer Page: https://duck.co/ia/view/{{ID}}

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mention

### CODE REVIEW

**Comment by daxtheduck** [comment]
@salmanfarisvp Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/pearl.json [hunk 1]
```diff
@@ -0,0 +1,240 @@
+{
+    "id": "pearl_cheat_sheet",
+    "name": "Pearl Language",
+    "description": "This is a handy reference, meant for beginning Perl programmers. ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: remove `.` present at the end of description

File: share/goodie/cheat_sheets/json/pearl.json [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        "sourceUrl": "http://www.pearl90.de/pearleingb.htm"
+    },
+    "aliases": [
+        "pearl programming",
+        "pearl language"
+    ],
+    "template_type": "code",
+    "section_order": [
+        "In-Built Functions",
+        "Operators and Loops ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: remove space from the end of `Operators and Loops`

File: share/goodie/cheat_sheets/json/pearl.json [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        "Matching and Regular Expression"
+
+    ],
+    "sections": {
+        "In-Built Functions": [{
+                "val": "Length of a string ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: same here, remove space!

File: share/goodie/cheat_sheets/json/pearl.json [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+                "key": "length($x)"
+            },
+            {
+                "val": "Remove a newline at the end of a string ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: same here, remove space..

File: share/goodie/cheat_sheets/json/pearl.json [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+                "key": "chomp($line);"
+            },
+            {
+                "val": "False if variable has never been set (or when you try to read past the end of an input file",
+                "key": "while (defined ($x=<> {code}"
+            },
+            {
+                "val": "Characters 3-7 of $long (first char is 0!) ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also, remove space

File: share/goodie/cheat_sheets/json/pearl.json [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+                "key": "$short=substr ($long, 2, 5)"
+            },
+            {
+                "val": "Add to end of array ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also, remove space

File: share/goodie/cheat_sheets/json/pearl.json [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+                "key": "push @arr, $x"
+            },
+            {
+                "val": "Remove last element of array, put in $x ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also, remove space

File: share/goodie/cheat_sheets/json/pearl.json [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+                "val": "Remove first element of an arrray, put in $x",
+                "key": "shift @arr;  (See also unshift)"
+            },
+            {
+                "val": "Number of things in the array",
+                "key": "$size = scalar @arr;"
+            },
+            {
+                "val": "split string->array, join array->string,  delete part of array, sort array in many way",
+                "key": "See also: split, join,  splice, sort  "
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also, remove space

File: share/goodie/cheat_sheets/json/pearl.json [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+            },
+            {
+                "val": "The lookup terms in the hash ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also, remove space

File: share/goodie/cheat_sheets/json/pearl.json [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+                "key": "@key = keys %hash"
+            },
+            {
+                "val": "See whether hash %hh has a value for key Hsp",
+                "key": "if (exists $hh{“Hsp”}) {...} "
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also, remove space

File: share/goodie/cheat_sheets/json/pearl.json [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+            },
+            {
+                "val": "Open outfile for writing, and associate it with filehandle HANDLE. Use “<infile” for reading ",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also, remove space

File: share/goodie/cheat_sheets/json/pearl.json [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+            {
+                "val": "Sets $i to 1 and does code. Sets $i to 2, … up to (and including) 100 ",
+                "key": "foreach my $i (1 .. 100) {code} (for and foreach are equivalent) "
+            },
+            {
+                "val": "Does code while the condition is true (If condition is false, never enters the loop.) ",
+                "key": "while ($a < $b) {code}"
+            }
+        ],
+        "Matching and Regular Expressions": [{
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: change it to `Matching and Regular Expressions` , because it's not that is mentioned above!

**Comment by amarlearning** [comment]
the cheat sheet is good :smiley:
but you should check and remove all tail end spaces!

**Comment by salmanfarisvp** [comment]
thank you

File: share/goodie/cheat_sheets/json/pearl.json (hunk 10 above)
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: there is still some space left!

File: share/goodie/cheat_sheets/json/pearl.json [hunk 13]
```diff
@@ -0,0 +1,240 @@
...
+                "val": "The lookup terms in the hash",
+                "key": "@key = keys %hash"
+            },
+            {
+                "val": "See whether hash %hh has a value for key Hsp",
+                "key": "if (exists $hh{“Hsp”}) {...} "
+            },
+            {
+                "val": "Open outfile for writing, and associate it with filehandle HANDLE. Use “<infile” for reading",
+                "key": "open(HANDLE, '>outfile') or die “Can’t open $outfile: $!\n' "
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also!

File: share/goodie/cheat_sheets/json/pearl.json [hunk 14]
```diff
@@ -0,0 +1,240 @@
...
+                "val": "Read a line from filehandle HANDLE, put in $x",
+                "key": "$x=<HANDLE>"
+            },
+            {
+                "val": "Stop reading/writing a previously opened file",
+                "key": "close(HANDLE);"
+            },
+            {
+                "val": "Exits the program",
+                "key": "exit; "
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: and here also!

File: share/goodie/cheat_sheets/json/pearl.json [hunk 15]
```diff
@@ -0,0 +1,240 @@
...
+            {
+                "val": "Sets $i to 1 and does code. Sets $i to 2, … up to (and including) 100",
+                "key": "foreach my $i (1 .. 100) {code} (for and foreach are equivalent)"
+            },
+            {
+                "val": "Does code while the condition is true (If condition is false, never enters the loop.)",
+                "key": "while ($a < $b) {code}"
+            }
+        ],
+        "Matching and Regular Expressions": [{
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: remove an extra `s` from here!
You have written `Matching and Regular Expression` in your section field and here you are using `Matching and Regular Expressions`. correct this!

File: share/goodie/cheat_sheets/json/pearl.json [hunk 16]
```diff
@@ -0,0 +1,240 @@
...
+            {
+                "val": "Global substitution.",
+                "key": "g"
+            },
+            {
+                "val": "Any one character(except \n)",
+                "key": "."
+            },
+            {
+                "val": "Any one of.",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: remove that `.` from the end of `val`

File: share/goodie/cheat_sheets/json/pearl.json [hunk 17]
```diff
@@ -0,0 +1,240 @@
...
+            {
+                "val": "Do a Substitution",
+                "key": "s///"
+            },
+            {
+                "val": "Ignore case.",
+                "key": "i"
+            },
+            {
+                "val": "Global substitution.",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: remove that `.` from the end of `val`

File: share/goodie/cheat_sheets/json/pearl.json [hunk 18]
```diff
@@ -0,0 +1,240 @@
...
+            {
+                "val": "Default variable",
+                "key": "$_"
+            },
+            {
+                "val": "Do a Substitution",
+                "key": "s///"
+            },
+            {
+                "val": "Ignore case.",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: remove that `.` from the end of `val`

File: share/goodie/cheat_sheets/json/pearl.json [hunk 19]
```diff
@@ -0,0 +1,240 @@
...
+                "val": "Are numbers different?",
+                "key": "if (1 != 2) {print 'diff num'}"
+            },
+            {
+                "val": "Are numbers greater than, less than, etc.",
+                "key": ">   <   >=   <="
+            },
+            {
+                "val": "Does text have exactly the same letters?",
+                "key": "if (“a” eq “a”) {print “same text”} "
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: remove that extra space!

File: share/goodie/cheat_sheets/json/pearl.json [hunk 20]
```diff
@@ -0,0 +1,240 @@
...
+
+            },
+            {
+                "val": "Exits the program with error message",
+                "key": "die 'Something broke!\n';"
+
+            }
+
+        ],
+        "Operators and Loops ": [{
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/pearl.json]: remove this extra space as well!

**Comment by amarlearning** [comment]
@salmanfarisvp you should change you IA page ID to `pearl_cheat_sheet` and also mention your IA page URL in your PR. where it is asked!

**Comment by salmanfarisvp** [comment]
Sorry ,  I didn't  get that

File: share/goodie/cheat_sheets/json/pearl.json [hunk 21]
```diff
@@ -0,0 +1,240 @@
+{
+    "id": "pearl_cheat_sheet",
+    "name": "Pearl Language",
+    "description": "This is a handy reference, meant for beginning Perl programmers. Not everything is mentioned",
+    "metadata": {
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/pearl.json]: Do you mean `Perl programmers`, or `Pearl programmers`?

**Comment by GuiltyDolphin** [comment]
@salmanfarisvp There is already a Perl cheat sheet (https://duck.co/ia/view/perl_cheat_sheet), perhaps you'd like to contribute to that instead?

**Pull request closed without merging**